### PR TITLE
Enhance readability of Minimax

### PR DESCRIPTION
**Pull request by Hardvan**
Added function docstring and comments explaining the base case, maximizer and minimizer turns.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: backtracking/minimax.py [hunk 1]
```diff
@@ -35,36 +50,45 @@ def minimax(
     12
     """
 
+    # Check for invalid inputs
```
Comment (tianyizheng02) [inline backtracking/minimax.py:53]:
```suggestion
```
I feel that this comment isn't really necessary. The if-statement is very simple so it's already pretty clear what it does.

File: backtracking/minimax.py [hunk 2]
```diff
@@ -35,36 +50,45 @@ def minimax(
...
     print(minimax(0, 0, True, scores, height))
 
 
 if __name__ == "__main__":
     import doctest
 
-    doctest.testmod()
-    main()
+    doctest.testmod()  # Run doctests to ensure the function behaves as expected
+    main()  # Run the main function to demonstrate the minimax algorithm in action
```
Comment (tianyizheng02) [inline backtracking/minimax.py:94]:
```suggestion
    doctest.testmod()
    main()
```
Likewise, I feel that this is too simple to need any comments

File: backtracking/minimax.py [hunk 3]
```diff
@@ -16,6 +16,21 @@ def minimax(
     depth: int, node_index: int, is_max: bool, scores: list[int], height: float
 ) -> int:
     """
+    This function implements the minimax algorithm,
+    which helps achieve the maximum score in a game
+    by checking all possible moves.
```
Comment (tianyizheng02) [inline backtracking/minimax.py:21]:
```suggestion
    This function implements the minimax algorithm, which helps achieve the optimal
    score for a player in a two-player game by checking all possible moves. If the player
    is the maximizer, then the score is maximized. If the player is the minimizer, then
    the score is minimized.
```
Clarifying the documentation

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Comment by Hardvan** [comment]
I have committed the requested changes.

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**